Using a libjemalloc Allocator for Better Memory Management

goetzrrGit opened the pull request:
* **Tickets addressed:** AERIE-000
* **Review:** By commit  <!-- Choose from: "by commit", "by file" -->
* **Merge strategy:** Merge (no squash)  <!-- Choose from: "merge (no squash)", "squash and merge" -->

## Description
We are still noticing a slow climb in the RSS for the sequencing server. @dandelany found this thread on Node about using a different allocator. https://github.com/nodejs/help/issues/1518#issuecomment-991798619

I have implemented the suggested fix and noticed during my 13 expansion runs with a Clippers plan, expansion logic, and dictionaries that the RSS doesn't climb past 3.6GB where as before I was getting past 4GB

goetzrrGit committed:
Using a libjemalloc allocator for better memory management.

* The docker container is Ubuntu LTS so we shouldn't have to worry about it until 2032

dandelany commented:
Related issue: #1476

skovati reviewed (commented):
FYI, alpine does support jemalloc as well:
https://pkgs.alpinelinux.org/packages?name=jemalloc&branch=v3.17&repo=main&arch=&maintainer=

Mythicaeda committed:
Use `node:bookworm-slim` instead of `ubuntu` image

- Size of final Ubuntu image: `652MB`
- Size of final Node:bookworm-slim image: `361MB`

Additionally, flexibly sets the path for `libjemalloc` based off the container's architecture.

Mythicaeda commented:
> FYI, alpine does support jemalloc as well: https://pkgs.alpinelinux.org/packages?name=jemalloc&branch=v3.17&repo=main&arch=&maintainer=

Documenting this here as well: I tried using the alpine version and didn't see it loaded when checking the node processes.

goetzrrGit commented:
@Mythicaeda I tested this locally and verified that jemalloc is being used. Thank you for figuring it out. I can't approve this PR as it is my own but if you don't have any more updates feel free to approve it.
